cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1862,6 +1862,90 @@ if(testing AND NO_CONNECTION)
   endif()
 endif()
 
+#---Check for ZeroMQ when building RooFit::MultiProcess--------------------------------------------
+
+if (roofit_multiprocess AND NOT MSVC)
+  if(NOT builtin_zeromq)
+    message(STATUS "Looking for ZeroMQ (libzmq)")
+    # Clear cache before calling find_package(ZeroMQ),
+    # necessary to be able to toggle builtin_zeromq and
+    # not have find_package(ZeroMQ) find builtin ZeroMQ.
+    foreach(suffix FOUND INCLUDE_DIR INCLUDE_DIRS LIBRARY LIBRARIES)
+      unset(ZeroMQ_${suffix} CACHE)
+    endforeach()
+
+    # Temporarily prefer config mode over module mode, so that a CMake-installed system version
+    # gets detected before looking for an autotools-installed system version (which the
+    # FindZeroMQ.cmake module does).
+    set(CMAKE_FIND_PACKAGE_PREFER_CONFIG_ORIGINAL_VALUE ${CMAKE_FIND_PACKAGE_PREFER_CONFIG})
+    set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)
+
+    if(fail-on-missing)
+      find_package(ZeroMQ REQUIRED)
+    else()
+      find_package(ZeroMQ)
+      if(NOT ZeroMQ_FOUND)
+        message(STATUS "ZeroMQ not found. Switching on builtin_zeromq option")
+        set(builtin_zeromq ON CACHE BOOL "Enabled because ZeroMQ not found (${builtin_zeromq_description})" FORCE)
+      endif()
+    endif()
+
+    # Reset default find_package mode
+    set(CMAKE_FIND_PACKAGE_PREFER_CONFIG ${CMAKE_FIND_PACKAGE_PREFER_CONFIG_ORIGINAL_VALUE})
+    unset(CMAKE_FIND_PACKAGE_PREFER_CONFIG_ORIGINAL_VALUE)
+
+    # only check symbol if the package was found (we may still reach this point when it wasn't found
+    # and fail-on-missing is deactivated)
+    if(ZeroMQ_FOUND)
+      check_symbol_exists(zmq_ppoll zmq.h ZeroMQ_HAS_PPOLL
+              CMAKE_REQUIRED_DEFINITIONS ZMQ_BUILD_DRAFT_API)
```

Comment:
Have you checked that this works? When I try to build ROOT without the builtin zeromq but with a new zeromq package on my system (it has `zmq_ppoll`), I still the the error:
```
ERRORDetected ZeroMQ version is too old (does not have zmq_ppoll). Aborting, because fail-on-missing was set.
```
But when I try to compile the `CheckSymbolExists.c` file that cmake produces manually:
```C++
#include <zmq.h>

int main(int argc, char** argv) {
   return ((int*)(&zmq_ppoll))[argc];
}
```
It works! I think somehow the `ZMQ_BUILD_DRAFT_API` flag might not me propagated correctly. Does it work for your system?

Reply:
I've pushed a fix for this, could you check if it works for you now?